src/go/rdctl/cmd/shell.go

```diff
@@ -99,3 +115,81 @@ func setupLimaHome() error {
 	os.Setenv("LIMA_HOME", candidatePath)
 	return nil
 }
+
+const restartDirective = "Either run 'rdctl start' or start the Rancher Desktop application first"
+
+func checkLimaIsRunning(commandName string) bool {
+	// Ignore error messages; none are expected here
+	output, err := exec.Command(commandName, "ls", "0", "--format", "{{.Status}}").CombinedOutput()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to run 'rdctl shell': %s\n", err)
+		return false
+	}
+	if strings.HasPrefix(string(output), "Running") {
+		return true
+	}
+
+	if output == nil || strings.Contains(string(output), "No instance matching 0 found") {
+		fmt.Fprintf(os.Stderr, "The Rancher Desktop VM needs to be created.\n%s.\n", restartDirective)
+	} else {
+		fmt.Fprintf(os.Stderr,
+			"The Rancher Desktop VM needs to be in state \"Running\" in order to execute 'rdctl shell', but it is currently in state \"%s\".\n%s.\n", strings.TrimRight(string(output), "\n"), restartDirective)
+	}
+	return false
+}
+
+func isUTF16(outputBytes []byte) bool {
+	if len(outputBytes) < 2 {
+		return false
+	}
+	if outputBytes[0] == 255 && outputBytes[1] == 254 {
+		// It's da BOM
+		return true
+	}
+	return outputBytes[1] == 0
+}
+
+func checkWSLIsRunning(distroName string) bool {
+	// Ignore error messages; none are expected here
+	outputBytes, err := exec.Command("wsl", "--list", "-v").CombinedOutput()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to run 'wsl -l': %s\n", err)
+		return false
+	}
+	var output []byte
+	if isUTF16(outputBytes) {
+		decoder := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewDecoder()
+		unicodeReader := transform.NewReader(bytes.NewReader(outputBytes), decoder)
+		output, err = ioutil.ReadAll(unicodeReader)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to convert utf16 to bytes: %s\n", err)
+			return false
+		}
+	} else {
+		output = outputBytes
+	}
+	isListed := false
+	targetState := ""
+	for _, line := range regexp.MustCompile(`\r?\n`).Split(string(output), -1) {
+		fields := regexp.MustCompile(`\s+`).Split(strings.TrimLeft(line, " \t"), -1)
```

Comment:
I'd say it's overkill, and there's no need for `unicode.IsSpace` because the data's been converted to utf-8, and is currently only spaces and tabs. We've already decided not to future-proof this code, but deal with breakage if/when it happens due to changes in the future.